projects/storefrontapp-e2e-cypress/cypress/integration/b2b/order-approval/b2b-order-approval.e2e-spec.ts

```diff
@@ -0,0 +1,234 @@
+import * as orderApproval from '../../../helpers/b2b/b2b-order-approval';
+import { POWERTOOLS_BASESITE } from '../../../sample-data/b2b-checkout';
+import * as sampleData from '../../../sample-data/b2b-order-approval';
+
+context('B2B - Order Approval', () => {
+  before(() => {
+    cy.window().then((win) => win.sessionStorage.clear());
+    Cypress.env('BASE_SITE', POWERTOOLS_BASESITE);
+  });
+
+  it('should display order approval details in order details page', () => {
+    orderApproval.loginB2bUser();
+    orderApproval.getPendingOrderDetails();
+
+    cy.visit(`/my-account/order/${sampleData.ORDER_CODE}`);
+    assertPermissionResults(sampleData.pendingOrder);
+  });
+
+  it('should display order approval list', () => {
+    orderApproval.loginB2bApprover();
+    orderApproval.getOrderApprovalList();
+    orderApproval.visitOrderApprovalListPage();
+
+    cy.get('cx-order-approval-list a.cx-order-approval-value')
+      .eq(0)
+      .should(
+        'contain',
+        sampleData.approvalOrderList.orderApprovals[0].order.code
+      );
+
+    cy.get('cx-order-approval-list a.cx-order-approval-value')
+      .eq(1)
+      .should('contain', sampleData.none);
+
+    cy.get('cx-order-approval-list a.cx-order-approval-value')
+      .eq(2)
+      .should(
+        'contain',
+        sampleData.approvalOrderList.orderApprovals[0].order.orgCustomer.name
+      );
+
+    cy.get('cx-order-approval-list a.cx-order-approval-value')
+      .eq(3)
+      .should('contain', sampleData.orderPlacedDate);
+
+    cy.get('cx-order-approval-list a.cx-order-approval-value')
+      .eq(4)
+      .should('contain', sampleData.statusPendingApproval);
+
+    cy.get('cx-order-approval-list a.cx-order-approval-value')
+      .eq(5)
+      .should(
+        'contain',
+        sampleData.approvalOrderList.orderApprovals[0].order.totalPrice
+          .formattedValue
+      );
+  });
+
+  it('Should display approval detail page', () => {
+    orderApproval.loginB2bApprover();
```

Comment:
Same as above. login can be done on beforeEach() or before() hook

Reply:
done, move to `beforeEach()`